### PR TITLE
add diagnostic items for clippy's `trim_split_whitespace`

### PR DESCRIPTION
**Pull request by FoseFx**
Adding the following diagnostic items:
 * str_split_whitespace,
 * str_trim,
 * str_trim_start,
 * str_trim_end

They are needed for https://github.com/rust-lang/rust-clippy/pull/8575

r? @flip1995

### CODE REVIEW

**Comment by flip1995** [comment]
Thanks! Naming of those diagnostic items is in accordance with the naming of the diagnostic items in the `mem` module, so this seems like a good naming for those functions.

@bors r+ rollup

**Comment by flip1995** [comment]
@bors r- 

Whoops too trigger happy on that r+

File: compiler/rustc_span/src/symbol.rs [hunk 1]
```diff
@@ -1347,6 +1347,10 @@ symbols! {
         store,
         str,
         str_alloc,
+        str_split_whitespace,
+        str_trim,
+        str_trim_start,
+        str_trim_end,
```
Comment (flip1995) [inline compiler/rustc_span/src/symbol.rs:1353]:
```suggestion
        str_trim_end,
        str_trim_start,
```

Comment (flip1995) [reply compiler/rustc_span/src/symbol.rs:1353]: Please squash this change in your existing commit.

Comment (FoseFx) [reply compiler/rustc_span/src/symbol.rs]: Yeah, I should learn the alphabet

**Pull request merged**